wit/types.wit

```diff
@@ -1,44 +1,43 @@
-interface messaging-types {
+interface types {
     /// A connection to a message-exchange service (e.g., buffer, broker, etc.).
     resource client {
         connect: static func(name: string) -> result<client, error>;
+        disconnect: func() -> result<_, error>;
     }
     
-    /// TODO(danbugs): This should be eventually extracted as an underlying type for other wasi-cloud-core interfaces.
-    resource error {
-        trace: static func() -> string;    
+    /// Errors that can occur when using the messaging interface.
+    variant error {
+        /// The request or operation timed out.
+        timeout,
+        /// An error occurred with the connection. Includes a message for additional context
+        connection(string),
+        /// A permission error occurred. Includes a message for additional context
+        permission-denied(string),
+        /// A catch all for other types of errors
+        other(string),
     }
   
-    /// There are two types of channels:
-    /// - publish-subscribe channel, which is a broadcast channel, and
-    /// - point-to-point channel, which is a unicast channel.
-    ///
-    /// The interface doesn't highlight this difference in the type itself as that's uniquely a consumer issue.
-    type channel = string;
-  
-    /// Configuration includes a required list of channels the guest is subscribing to, and an optional list of extensions key-value pairs 
-    /// (e.g., partitions/offsets to read from in Kafka/EventHubs, QoS etc.).
-    record guest-configuration {
-        channels: list<channel>,
-        extensions: option<list<tuple<string, string>>>
-    }
-  
-    /// Format specification for messages 
-    ///  - more info: https://github.com/clemensv/spec/blob/registry-extensions/registry/spec.md#message-formats
-    ///  - message metadata can further decorate w/ things like format version, and so on.
-    enum format-spec {
-        cloudevents,
-        http,
-        amqp,
-        mqtt,
-        kafka,
-        raw
-    }
-  
-    /// A message with a binary payload, a format specification, and decorative metadata.
-    record message {
-        data: list<u8>,
-        format: format-spec,
-        metadata: option<list<tuple<string, string>>>
+    /// A message with a binary payload and additional information
+    resource message {
+        constructor(topic: string, data: list<u8>);
+        /// The topic/subject/channel this message was received or should be sent on
+        topic: func() -> string;
+        /// Set the topic/subject/channel this message should be sent on
+        set-topic: func(topic: string);
```

Comment:
I needed to clarify this, but I never said anything; I self-reply. Could it be an ambiguous language problem?

Is the intent that such a Message is the very last data structure in the component, so you must say where it is going at that point?

Reply:
Just to clarify: I am advocating to add the `topic` to the `send` function, instead of the message. I believe that the topic is part of the publishing process, and not the message. The message should be able to stand by itself.

That said, keeping the `topic` in the message makes sense in the context of provenance.

Reply:
I'll defer this to @thomastaylor312 —do you want to elaborate a bit more on the reason behind the change of having topic be a part of `message` instead of an argument to `send`?

Reply:
I thought that,
To make a point, I see `message` as a `send-command` single argument object, so such a command object has all the information for a given use case (send) to be done.
I know it is confusing, because `message`

Reply:
Yep, I think @jocrau is correct here. I think the best course would be to remove `set-topic` from the message resource and then have topic be an argument on `send`

Reply:
Does that mean anything related to `topic` would be removed from the `message` then?

Reply:
I think it is still good to have the getter for topic (for usage w/ `handle`), but it should not be required in the constructor anymore.

Reply:
Added `topic` to `send`—`request-reply` pattern is untouched. You can still get `topic` metadata from `message` but no longer call `set-topic` on a message resource.